feat: multi chain latestEvent query and add collection name

preschian opened the pull request:
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3945
- [x] add collection name in latestEvent query
- [x] request to multi chain queries

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
<img width="1624" alt="Screen Shot 2022-09-23 at 16 51 00" src="https://user-images.githubusercontent.com/734428/191935676-4ca165dc-04b8-4b92-b393-ced2a7e25fa8.png">

roiLeo reviewed (commented):
Hey, I think this is breaking current landing page could you check?

preschian commented:
> Hey, I think this is breaking current landing page could you check?

aahh, my bad, I forgot that the old one already uses composable

roiLeo reviewed (commented):
Code look good, what would happend if one of the snek/subsquid endpoint went off? will it break the component/page?

note: would it be possible to change the text of the chain to be capitalized or uppercase? (snek should be Basilisk?)

petersopko commented:
@preschian links from the carousel lead to the selected chain (i.e. clicking at newest snek item links to rmrk/gallery/...)

petersopko reviewed (changes requested):
links pls, otherwise lgtm

preschian commented:
> Code look good, what would happend if one of the snek/subsquid endpoint went off? will it break the component/page?
> 
> note: would it be possible to change the text of the chain to be capitalized or uppercase? (snek should be Basilisk?)

oh, good points. I updated the code, so should be not broke carousel component if one of the query throw error


> @preschian links from the carousel lead to the selected chain (i.e. clicking at newest snek item links to rmrk/gallery/...)

oops, fixed 🙏🏻

petersopko commented:
wfm now, I guess our spotlight/curated list will get update once Basilisk goes live 👍🏻

Jarsen136 commented on components/carousel/module/CarouselInfo.vue:
```diff
@@ -26,12 +32,19 @@ import type { CarouselNFT } from '@/components/base/types'
+const chainName = computed(() => {
+  const name = {
+    rmrk: 'RMRK',
+    snek: 'Basilisk',
```
Why renaming `snek`nft with `Basilisk` 👀

preschian replied:
tbh, I still don't know the difference 🙈
what was the correct name?

Jarsen136 replied:
@petersopko  Could you pls answer this question?

petersopko replied:
I think we are going to be eventually sunsetting /snek route as this is only the final testnet where we could mirror the functionality of the actual Basilisk which is coming in live at 
- #4030

roiLeo replied:
From https://github.com/kodadot/nft-gallery/pull/4011#pullrequestreview-1119281239
> note: would it be possible to change the text of the chain to be capitalized or uppercase? (snek should be Basilisk?)

Since it's going to be on Basilisk parachain wouldn't it be wiser to rename it?

petersopko replied:
> Since it's going to be on Basilisk parachain wouldn't it be wiser to rename it?

it is Basilisk now, that's what @Jarsen136 was asking about
![image](https://user-images.githubusercontent.com/73316633/192493456-d5008fb5-7a9e-4c4a-b6e9-e3c59eaf59f9.png)

Jarsen136 commented on components/carousel/utils/useCarousel.ts:
```diff
@@ -60,6 +61,52 @@ export const useCarouselNftEvents = ({ type }: Types) => {
+    queryPrefix: 'subsquid',
+    queryName: 'lastNftListByEvent',
+    variables: nftEventVariables[type],
+    clientName: chain === 'rmrk' ? 'subsquid' : chain,
```
this utils function would help 

<img width="512" alt="image" src="https://user-images.githubusercontent.com/31397967/192282675-e9fc400f-cf6c-4b2f-b86f-8108e1145482.png">

preschian replied:
thanks 👍

petersopko commented:
@Jarsen136 can we check the code here and move forward? 🙏🏻

Jarsen136 reviewed (approved):
code lgtm

petersopko commented:
@yangwao payout here failed, I guess autocode is still having trouble 👀
